### PR TITLE
Updating worker flags gym.

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
- Move instructions from the gym itself into the user manual and updated to make it clearer.
- Updated the old launch config that was being used.

### CODE REVIEW

**Review by improbable-danny — APPROVED** [review]

**Review by improbable-valentyn — APPROVED** [review]

File: USER_MANUAL.md [hunk 1]
```diff
@@ -153,7 +153,14 @@ The ReplicatedStartupActorTest is failing, pending https://improbableio.atlassia
 * Tests a fix for UNR-1259: Fix of the WorkerFlags data structure not being per worker. When running through Unreal Editor using single process, different worker types can read other worker type's flags. As a result flags of different worker types with the same name get the wrong value.
 * Demonstrates that ClientWorkers and UnrealWorker read the correct value for the "test" worker flag, when both types have a "test" flag with different value.
 * Validation:
-  1. Use workerflags_testgym_config launch config to run multiplayer through the editor, for every worker running, different values based on their worker type should get printed/logged.
+  1. Navigate to: Project Settings->SpatialOS GDK for Unreal->Editor Settings->Launch.
```
Comment (oblm) [inline USER_MANUAL.md:156]: make this list:
1.
1.
1.

It will render as 

1.
2.
3.

this makes editing easier.

File: USER_MANUAL.md [hunk 2]
```diff
@@ -153,7 +153,14 @@ The ReplicatedStartupActorTest is failing, pending https://improbableio.atlassia
...
+  2. Ensure "Auto-generate launch configuration file" is disabled.
+  3. Set "Launch configuration file path" to "workerflags_testgym_config.json" (the file exists within the test gyms project).
+  4. Navigate to: Editor Preferences->Level Editor->Play->Multiplayer Options.
+  5. Ensure "Run Under One Process" is enabled.
+  6. Play in editor.
```
Comment (oblm) [inline USER_MANUAL.md:161]:
```suggestion
  6. Play in editor with one client.
```

File: USER_MANUAL.md (hunk 1 above)
Comment (oblm) [inline USER_MANUAL.md:156]: As the first step, tell me what map to open.

File: USER_MANUAL.md [hunk 3]
```diff
@@ -153,7 +153,14 @@ The ReplicatedStartupActorTest is failing, pending https://improbableio.atlassia
...
+  7. Check that the server prints out the corresponding flag value (15 by default).
```
Comment (oblm) [inline USER_MANUAL.md:162]: 1. Where do I check this, in the Output log window?
2. Corresponding to what?
3. Is the default value stored in `workerflags_testgym_config.json`?

File: USER_MANUAL.md [hunk 4]
```diff
@@ -153,7 +153,14 @@ The ReplicatedStartupActorTest is failing, pending https://improbableio.atlassia
 * Tests a fix for UNR-1259: Fix of the WorkerFlags data structure not being per worker. When running through Unreal Editor using single process, different worker types can read other worker type's flags. As a result flags of different worker types with the same name get the wrong value.
 * Demonstrates that ClientWorkers and UnrealWorker read the correct value for the "test" worker flag, when both types have a "test" flag with different value.
 * Validation:
-  1. Use workerflags_testgym_config launch config to run multiplayer through the editor, for every worker running, different values based on their worker type should get printed/logged.
+  1. Navigate to: Project Settings->SpatialOS GDK for Unreal->Editor Settings->Launch.
+  2. Ensure "Auto-generate launch configuration file" is disabled.
+  3. Set "Launch configuration file path" to "workerflags_testgym_config.json" (the file exists within the test gyms project).
```
Comment (oblm) [inline USER_MANUAL.md:158]:
```suggestion
  3. Set "Launch configuration file path" to `workerflags_testgym_config.json` (the file exists within the test gyms project).
```

File: USER_MANUAL.md [hunk 5]
```diff
@@ -153,7 +153,14 @@ The ReplicatedStartupActorTest is failing, pending https://improbableio.atlassia
...
+  4. Navigate to: Editor Preferences->Level Editor->Play->Multiplayer Options.
+  5. Ensure "Run Under One Process" is enabled.
+  6. Play in editor.
+  7. Check that the server prints out the corresponding flag value (15 by default).
+  8. Check that the client prints out the corresponding flag value (5 by default).
```
Comment (oblm) [inline USER_MANUAL.md:163]:
```suggestion
  8. Check that the same flag value (5 by default) is printed in the client viewport.
```

File: spatial/workerflags_testgym_config.json [hunk 6]
```diff
@@ -1,67 +1,54 @@
...
+                "manual_worker_connection_only": true
+            }
+        },
         {
-          "all": {}
+            "worker_type": "UnrealClient",
+            "flags": [
+                {
+                  "name": "test",
+                  "value": "5"
```
Comment (oblm) [inline spatial/workerflags_testgym_config.json:35]: Aha I guess this is the default value, you should link to this line in the user manual to make the connection clear.

**Review by oblm — CHANGES_REQUESTED** [review]

File: spatial/workerflags_testgym_config.json (hunk 6 above)
Comment (MatthewSandfordImprobable) [reply spatial/workerflags_testgym_config.json:35]: I think referencing the file from the USER_MANUAL should be adequate. I think referencing lines numbers has proven to be fragile in the past.

**Review by oblm — APPROVED** [review]
great work

**Pull request merged**